src/adv_diff/BrinkmanAdvDiffSemiImplicitHierarchyIntegrator.cpp

```diff
@@ -104,14 +104,6 @@ BrinkmanAdvDiffSemiImplicitHierarchyIntegrator::BrinkmanAdvDiffSemiImplicitHiera
     bool register_for_restart)
     : AdvDiffSemiImplicitHierarchyIntegrator(object_name, input_db, register_for_restart)
 {
-#if !defined(NDEBUG)
-    TBOX_ASSERT(!object_name.empty());
-    TBOX_ASSERT(input_db);
-#endif
```

Comment:
Why was this removed?

Reply:
Because these conditions are checked in the ```AdvDiffSemiImplicitHierarchyIntegrator``` class constructor. Here it was redundant.

Reply:
I think it is better to keep the assertions if they are relevant for the current method, even if they are checked elsewhere --- they shouldn't cause any harm here. Also, although it seems unlikely, they also could (accidentally or intentionally) be removed from the other implementation.

Reply:
Sounds good.